chainer/_link.py

```diff
@@ -0,0 +1,373 @@
+import sys
+import typing as tp  # NOQA
+import warnings
+
+import numpy
+
+import chainer
+from chainer import backend
+from chainer.backends import _cpu
+from chainer.backends import cuda
+from chainer.backends import intel64
+from chainer import types  # NOQA
+import chainerx
+
+
+class DeviceResidentMixin(object):
+    """A mix-in class to support multi-device hierarchy.
+    """
```

Comment:
Put a blank line before the docstring. I think it's also ok to put the closing quotes at the same line (not mandatory, though):
```py3
class DeviceResidentMixin(object):

    """A mix-in class to support multi-device hierarchy."""
```

Reply:
Fixed as suggested.